react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,44 @@
+import './App';
+import { ImgCard } from './ImgCard';
+import { Prev } from './Prev';
+import { Next } from './Next';
+import { Dots } from './Dots';
+import { useEffect, useState } from 'react';
+
+type ImgProp = {
+  src: string;
+  alt: string;
+};
+
+type Prop = {
+  details: ImgProp[];
+};
+
+export function Carousel({ details }: Prop) {
+  const [cycle, setCycle] = useState(0);
+  function handlePrev(): void {
+    setCycle((cycle - 1 + details.length) % details.length);
+  }
+
+  function handleNext(): void {
+    setCycle((cycle + 1) % details.length), [details.length, cycle];
```

Comment:
It looks like you are trying to pass a dependency array to `setCycle`, but it doesn't accept one. I think you were confused with useCallback, which you should use for this function.